flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorSubmissionTest.java

```diff
@@ -504,26 +441,11 @@ public void testUpdateTaskInputPartitionsFailure() throws Exception {
 	 */
 	@Test(timeout = 10000L)
 	public void testLocalPartitionNotFound() throws Exception {
-		final ExecutionAttemptID eid = new ExecutionAttemptID();
-
-		final IntermediateDataSetID resultId = new IntermediateDataSetID();
-		final ResultPartitionID partitionId = new ResultPartitionID();
-
-		final ResultPartitionLocation loc = ResultPartitionLocation.createLocal();
-
-		final InputChannelDeploymentDescriptor[] inputChannelDeploymentDescriptors =
-			new InputChannelDeploymentDescriptor[] {
-				new InputChannelDeploymentDescriptor(partitionId, loc)};
-
-		final InputGateDeploymentDescriptor inputGateDeploymentDescriptor =
-			new InputGateDeploymentDescriptor(resultId, ResultPartitionType.PIPELINED, 0, inputChannelDeploymentDescriptors);
-
-		final TaskDeploymentDescriptor tdd =
-			createTestTaskDeploymentDescriptor("Receiver",
-				eid,
-				Tasks.AgnosticReceiver.class,
-				1, Collections.emptyList(),
-				Collections.singletonList(inputGateDeploymentDescriptor));
+		ResourceID producerLocation = new ResourceID("local");
+		DefaultShuffleDeploymentDescriptor sdd =
+			createSddWithLocalConnection(new IntermediateResultPartitionID(), producerLocation, 10000);
+		TaskDeploymentDescriptor tdd = createReceiver(sdd, producerLocation);
```

Comment:
why does the receiver get the producer location assigned?

Reply:
it should result in identifying that the partition is located locally in SingleInputGateFactory on the consumer side, by comparing producer and consumer resource ids.